android/app/src/main/java/org/electroncash/electroncash3/NewWallet.kt

```diff
@@ -19,6 +25,11 @@ import kotlin.properties.Delegates.notNull
 val libKeystore by lazy { libMod("keystore") }
 val libWallet by lazy { libMod("wallet") }
 
+val keystores by lazy { ArrayList<PyObject>() }
```

Comment:
It isn't safe to use a global variable to track dialog state, because it's too easy for it to get detached from the dialog lifecycle. I was able to break this in three different ways when creating a 2of2 wallet:

* Enter the first key, then when the MasterPublicKeyDialog appears, rotate the screen. This causes NewWalletDialog1.onDismiss to be called, which clears `keystores`. Then after you enter the second key, it tries to create the wallet with only one keystore, which causes the following crash:

      com.chaquo.python.PyException: BaseException: wallet format requires update
        at <python>.electroncash.keystore.load_keystore(keystore.py:677)
        at <python>.electroncash.wallet.load_keystore(wallet.py:3165)
        at <python>.electroncash.wallet.load_keystore_wrapper(wallet.py:337)
        at <python>.electroncash.wallet.__init__(wallet.py:286)
        at <python>.electroncash.wallet.__init__(wallet.py:2968)
        at <python>.electroncash.wallet.__init__(wallet.py:3145)
        at <python>.electroncash_gui.android.console.create_multisig(console.py:184)

* Enter the first key, then when the MasterPublicKeyDialog appears, press Home to send the app to the background, then kill it with the red Terminate button in the Android Studio Logcat window. Because the app went into the background before it was killed, it had a chance to save its state, including the dialog arguments. So when you re-launch the app, the dialogs will re-appear as they were before. But it's a new process, so `keystores` is empty, and attempting to continue will lead to the same crash as above.

* Enter the first key (let's call it A), then when the MasterPublicKeyDialog appears, press Back and enter a different key B. Then enter the second key C. You should get a wallet with keys B and C, but you actually get one with keys A and B.

I think the easiest fix is to move the keystores to the dialog arguments: 
* Make `create` in console.py return the keystore as a JSON string.
* Pass those strings from dialog to dialog in a StringArrayList dialog argument. That would make `i_signer` unnecessary, because it would be implied by the size of the array. It's probably safer to make a new array for each dialog rather than modifying the existing one, otherwise it could cause confusion when the user presses Back.
* Finally, make `create_multisig` convert the JSON strings back into dicts before putting them into the wallet storage.